File: ncert/9/3/23/codes/main.c




#include <stdio.h>
#include <math.h>

double binomialCoeff(int n, int k) {
    double result = 1.0;
    for (int i = 1; i <= k; i++) {
        result *= (double)(n - i + 1) / i;
    }
    return result;
}

int main() {
    int n = 5;        
    double p = 0.5;   
    int k = 3;        

    double binomial_pmf = binomialCoeff(n, k) * pow(p, k) * pow(1 - p, n - k);
    double mu = n * p;
    double sigma = sqrt(n * p * (1 - p));

    double normal_pdf = exp(-0.5 * pow((k - mu) / sigma, 2)) / (sigma * sqrt(2 * M_PI));
    double normal_ans1 = exp(-0.5 * pow((k - mu) / sigma, 2)) / (sigma * sqrt(2 * M_PI));

    printf("answer through normal dist: %lf\n", normal_ans1);
    printf("answer through binomial dist: %lf\n", binomial_pmf);

    return 0;
}

File: ncert/9/3/23/codes/main.py
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import binom, norm

n = 5 
#no of items
p = 0.5  
#probability of defective items
k = np.arange(0, n+1)  
binomial_pmf = binom.pmf(k, n, p)
mu = n * p
sigma = np.sqrt(n * p * (1 - p))
x = np.linspace(0, n, 100000)
normal_pdf = norm.pdf(x, mu, sigma)
normal_ans1=norm.pdf((1-mu)/sigma,mu,sigma)
print("answer through normal dist: ",normal_ans1)
binomial_ans2=binom.pmf(1,n,p)
print("answer through binomial dist: ",binomial_ans2)
binom_ans2 = binom.pmf(1, n, p)
plt.stem(k, binomial_pmf, label='PMF', basefmt='b-')
plt.plot(x, normal_pdf, label='PDF', color='r')
plt.axvline(x=1, color='g', linestyle='--', label='x=3')
plt.xlabel('Number of Defective Articles')
plt.ylabel('Probability')
plt.legend()
plt.savefig("./figs/fig1.png")
